scripts/onnx_generator/OnnxWrapper.py

```diff
@@ -0,0 +1,490 @@
+    
+    def __str__(self):
+        return self.text()
+
+
+class OnnxInputList(list):
+    def __init__(self, schema):
+        super()
+        self.extend([ OnnxInput(i) for i in schema.inputs])
+    
+    def text(self, prefix=""):
+        paragraphs = [ i.text(prefix) for i in self ]
+        return f"\n{prefix}\n".join(paragraphs)
+    
+    def __str__(self):
+        return self.text()
+
+
+class OnnxOutput():
+    def __init__(self, output):
+        if isinstance(output, dict):
+            self.name = output['name']
+            self.description = output['description']
+            self.isHomogeneous = output['isHomogeneous']
+            self.option = output['option']
+            self.constraint = output['constraint']
+            self.types = output['types']
+        else:
+            self.name = output.name
+            self.description = output.description
+            self.isHomogeneous = output.isHomogeneous
+            self.option = output.option.name
+            self.constraint = output.typeStr
+            self.types = OnnxTypeList(output.types)
+    
+    def text(self, prefix=""):
+        lines = []
+        lines.append(f"{prefix}Output {self.constraint} {self.name}:")
+        lines.append(format_text(prefix + "  ", None, [self.description]))
+        lines.append(format_text(prefix + "  ", "Allowed Types:", [str(self.types)] ))
+        return "\n".join(lines)
+
+    def __repr__(self):
+        return self.__dict__.__repr__()
+    
+    def __str__(self):
+        return self.text()
+
+class OnnxOutputList(list):
+    def __init__(self, schema):
+        super()
+        self.extend([ OnnxOutput(i) for i in schema.outputs])
+    
+    def text(self, prefix=""):
+        paragraphs = [ i.text(prefix) for i in self ]
+        return f"\n{prefix}\n".join(paragraphs)
+    
+    def __str__(self):
+        return self.text()
+
+class OnnxDoc():
+    def __init__(self, schema):
+        if isinstance(schema,str):
+            self.doc = schema
+        else:
+            self.doc = schema.doc.strip()
+
+    def __repr__(self):
+        return f"OnnxDoc({self.doc.__repr__()})"
+    
+    def text(self, prefix=""):
+        return format_text(prefix, None, [self.doc])
+    
+    def __str__(self):
+        return self.text()
+
+class OnnxSchema():
+
+    def __init__(self, schema):
+        self.name = None
+        self.doc = None
+        self.deprecated = None
+        self.operator_name = None
+        self.version = None
+        self.domain = None
+        self.constraints = None
+        self.attributes = None
+        self.inputs = None
+        self.outputs = None
+        self.ref_doc = None
+        self.range_input = None
+        self.range_output = None
+        self.ref_file = None
+        if isinstance(schema, dict):
+            self.__dict__.update(schema)
+        else:
+            self.doc = OnnxDoc(schema)
+            self.name = schema.name
+            self.deprecated = schema.deprecated
```

Comment:
Just FYI, `deprecated` doesn't exist in some older versions (i.e. 1.0.0)

Reply:
good to know. thanks